dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/passes/reachingdef/ReachingDefPass.scala

```diff
@@ -125,6 +127,35 @@ class ReachingDefPass(cpg: Cpg, maxNumberOfDefinitions: Int = 4000) extends Para
           addEdge(method, node)
         }
       }
+
+    // Add edges for blocks used as arguments
+    allNodes
+      .collect { case c: Call => c }
+      .foreach { call =>
+        call.argument.isBlock.foreach { block =>
+          block.astChildren.lastOption match {
+            case None => // Do nothing
+
+            case Some(node: Identifier) =>
+              val edgesToAdd = in(node).toList.flatMap { inDef =>
+                numberToNode(inDef) match {
+                  case identifier: Identifier => Some(identifier)
+                  case _                      => None
+                }
+              }
+              edgesToAdd.foreach { inNode =>
+                addEdge(inNode, block, nodeToEdgeLabel(inNode))
+              }
+              if (edgesToAdd.nonEmpty) {
+                addEdge(block, call)
+              }
+
+            case Some(node: Call) =>
+              addEdge(node, call, nodeToEdgeLabel(node))
+              addEdge(block, call)
+          }
+        }
+      }
```

Comment:
This turned into a monolithic PR of stuff found while fixing javasrc2cpg constructors. This code snippet is the most important one that may impact other frontends.

Reply:
Do we have to handle other types of nodes that may appear as last AST child in a block?
Otherwise, it may crash with a match error.

Reply:
Ah, good catch! There are cases such as `return` and the various control structures that would break this.